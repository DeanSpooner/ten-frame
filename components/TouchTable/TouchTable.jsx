import { useState } from "react";
import { TouchableOpacity, View } from "react-native";
import touchTableStyles from "./TouchTable.styles";

const TouchTable = ({ countersTouched, setCountersTouched }) => {
  const [touchedCountersArray, setTouchedCountersArray] = useState([]);

  const lengthOfFrame = 10;
  const arrayOfCounters = Array.from(Array(lengthOfFrame)).map((e, i) => i + 1);

  const addToArray = (num) => {
    setTouchedCountersArray([...touchedCountersArray, num]);
    setCountersTouched(countersTouched + 1);
  };

  const removeFromArray = (num) => {
    setTouchedCountersArray((item) =>
      item.filter((element) => {
        return element !== num;
      })
    );
    setCountersTouched(countersTouched - 1);
  };

  return (
    <View style={touchTableStyles.container}>
      {arrayOfCounters.map((num) => (
        <TouchableOpacity
          style={touchTableStyles.cell}
          key={num}
          onPress={() => {
            if (touchedCountersArray.includes(num)) {
              removeFromArray(num);
            } else {
              addToArray(num);
            }
          }}
        >
          <View
            style={
              touchedCountersArray.includes(num)
                ? [touchTableStyles.counter]
                : ""
            }
          ></View>
        </TouchableOpacity>
      ))}
    </View>
  );
};

export default TouchTable;
