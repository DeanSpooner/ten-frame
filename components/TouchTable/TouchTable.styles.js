import { StyleSheet } from "react-native";

export default touchTableStyles = StyleSheet.create({
  container: {
    width: "100%",
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "center",
    maxWidth: 360,
  },
  cell: {
    backgroundColor: "#fff",
    borderColor: "#000",
    borderWidth: 3,
    height: 72,
    width: 72,
    justifyContent: "center",
    alignItems: "center",
  },
  counter: {
    width: 65,
    height: 65,
    backgroundColor: "#f00",
    borderColor: "#000",
    borderWidth: 3,
    borderRadius: 50,
    overflow: "hidden",
  },
});
