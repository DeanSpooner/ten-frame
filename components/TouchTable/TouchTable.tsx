import React, { useState } from "react";
import { TouchableOpacity, View } from "react-native";
import touchTableStyles from "./TouchTable.styles";

interface TouchTableProps {
  countersTouched: number;
  setCountersTouched: React.Dispatch<React.SetStateAction<number>>;
}

const TouchTable: React.FC<TouchTableProps> = ({
  countersTouched,
  setCountersTouched,
}) => {
  const [touchedCountersArray, setTouchedCountersArray] = useState<number[]>(
    []
  );

  const lengthOfFrame: number = 10;
  const arrayOfCounters: number[] = Array.from(Array(lengthOfFrame)).map(
    (e, i) => i + 1
  );

  const addToArray = (num: number) => {
    setTouchedCountersArray([...touchedCountersArray, num]);
    setCountersTouched(countersTouched + 1);
  };

  const removeFromArray = (num: number) => {
    setTouchedCountersArray((item) =>
      item.filter((element) => {
        return element !== num;
      })
    );
    setCountersTouched(countersTouched - 1);
  };

  return (
    <View style={touchTableStyles.container}>
      {arrayOfCounters.map((num) => (
        <TouchableOpacity
          style={touchTableStyles.cell}
          key={num}
          onPress={() => {
            if (touchedCountersArray.includes(num)) {
              removeFromArray(num);
            } else {
              addToArray(num);
            }
          }}
        >
          <View
            //@ts-ignore
            style={
              touchedCountersArray.includes(num)
                ? [touchTableStyles.counter]
                : ""
            }
          ></View>
        </TouchableOpacity>
      ))}
    </View>
  );
};

export default TouchTable;
