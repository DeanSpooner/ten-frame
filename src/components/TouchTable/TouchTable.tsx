import React, { useState } from "react";
import { TouchableOpacity, View } from "react-native";
import { arrayMaker } from "../../helpers/arrayMaker";
import touchTableStyles from "./TouchTable.styles";

interface TouchTableProps {
  countersTouched: number;
  setCountersTouched: React.Dispatch<React.SetStateAction<number>>;
}

const TouchTable: React.FC<TouchTableProps> = ({
  countersTouched,
  setCountersTouched,
}) => {
  const [touchedCountersArray, setTouchedCountersArray] = useState<number[]>(
    []
  );

  const lengthOfFrame: number = 10;

  const addToArray = (num: number) => {
    setTouchedCountersArray([...touchedCountersArray, num]);
    setCountersTouched(countersTouched + 1);
  };

  const removeFromArray = (num: number) => {
    setTouchedCountersArray((item) =>
      item.filter((element) => {
        return element !== num;
      })
    );
    setCountersTouched(countersTouched - 1);
  };

  return (
    <View style={touchTableStyles.container}>
      {arrayMaker(lengthOfFrame).map((num) => (
        <TouchableOpacity
          style={touchTableStyles.cell}
          key={num}
          onPress={() => {
            if (touchedCountersArray.includes(num)) {
              removeFromArray(num);
            } else {
              addToArray(num);
            }
          }}
        >
          <View
            //@ts-ignore
            style={
              touchedCountersArray.includes(num)
                ? [touchTableStyles.counter]
                : ""
            }
          ></View>
        </TouchableOpacity>
      ))}
    </View>
  );
};

export default TouchTable;
