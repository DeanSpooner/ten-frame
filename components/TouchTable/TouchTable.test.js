import React from "react";
import renderer from "react-test-renderer";
import TouchTable from "./TouchTable";

describe("<Table />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<TouchTable />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has 10 children", () => {
    const tree = renderer.create(<TouchTable />).toJSON();
    expect(tree.children.length).toBe(10);
  });
});
