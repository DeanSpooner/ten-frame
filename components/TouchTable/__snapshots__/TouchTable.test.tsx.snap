// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "maxWidth": 360,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "opacity": 1,
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "opacity": 1,
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "opacity": 1,
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "opacity": 1,
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "opacity": 1,
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "opacity": 1,
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "opacity": 1,
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "opacity": 1,
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "opacity": 1,
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "opacity": 1,
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
</View>
`;
